R/common-find-variation.R

```diff
@@ -425,4 +425,498 @@ getPipedValue <- function(string) {
+#'
+#' categoricalCols <- c("Dept","Gender")
+#' measureColumn <- c("LOS1","LOS2")
+#' 
+#' variationAcrossGroups(df = df, 
+#'                        categoricalCols = categoricalCols,
+#'                        measureColumn = measureColumn)
+#'                        
+#' treatment = c(rep("A", 20) , rep("B", 20) , rep("C", 20), rep("D", 20) ,  rep("E", 20))
+#' value = c( sample(2:5, 20 , replace = TRUE) , sample(6:10, 20 , replace = TRUE), 
+#'         sample(1:7, 20 , replace = TRUE), sample(3:10, 20 , replace = TRUE) , 
+#'         sample(10:20, 20 , replace = TRUE) )
+#' df1 = data.frame(treatment,value)
+#' 
+#' variationAcrossGroups(df1, categoricalCols = "treatment", measureColumn = "value")
+#'
+
+
+variationAcrossGroups <- function(df, 
+                                  categoricalCols,
+                                  measureColumn,
+                                  printPlot = TRUE,
+                                  printTable = TRUE,
+                                  dateCol = NULL) {
+  
+  if (!all(c(categoricalCols,measureColumn,dateCol) %in% names(df))) {
+    stop('The measure column or one of the categorical cols is not in the df')
+  }
+  
+  # Check that measure columns exist and are of proper type
+  if (length(measureColumn) > 1) {
+    for (i in 1:length(measureColumn)) {
+      if (class(df[[measureColumn[i]]]) != "numeric" &&  
+          class(df[[measureColumn[i]]]) != "integer") {
+        stop("measureColumn needs to be of class numeric or integer. ",
+             measureColumn[i], " appears to be of type ", 
+             class(df[[measureColumn[i]]]))
+      }
+    }
+  } else if (length(measureColumn) == 1) {
+    if (class(df[[measureColumn]]) != "numeric" &&  
+        class(df[[measureColumn]]) != "integer") {
+      stop("measureColumn needs to be of class numeric or integer. ",
+           measureColumn, " appears to be of type ", 
+           class(df[[measureColumn]]))
+    }
+  }
+  
+  # Check that categorical columns exist and are of proper type
+  if (length(categoricalCols) > 1) {
+    for (i in 1:length(categoricalCols)) {
+      if (class(df[[categoricalCols[i]]]) == "numeric" ||  
+          class(df[[categoricalCols[i]]]) == "integer") {
+        stop("categoricalCols cannot be of class numeric or integer. ", 
+             categoricalCols[i], " appears to be of type ", 
+             class(df[[categoricalCols[i]]]))
+      }
+    }
+  } else if (length(categoricalCols) == 1) {
+    if (class(df[[categoricalCols]]) == "numeric" ||  
+        class(df[[categoricalCols]]) == "integer") {
+      stop("categoricalCols cannot be of class numeric or integer. ", 
+           categoricalCols, " appears to be of type ", 
+           class(df[[categoricalCols]]))
+    }
+  }
+  
+  # Check if there are two many interactions
+  if (length(categoricalCols) > 10) {
+    stop("There are two many interactions. Length of categoricalCols cannot larger than 10.")
```

Comment:
Is 10 too few? What does the plot look like with (for example) 50 interactions? If there are more than the specified, will it still return the table?